sechub-scan/src/main/java/com/daimler/sechub/adapter/DefaultExecutorConfigSupport.java

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * A standard executor configuration support. Supports environment entry evaluation, simple key value checks etc.
+ * 
+ * @author Albert Tregnaghi
+ *
+ */
+public class DefaultExecutorConfigSupport {
+
+    private static final String ENV_PREFIX_ID = "env:";
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultExecutorConfigSupport.class);
+
+    protected Map<String, String> configuredExecutorParameters = new TreeMap<>();
+
+    protected ProductExecutorConfig config;
+    private SystemEnvironment systemEnvironment;
+
+    public DefaultExecutorConfigSupport(ProductExecutorConfig config, SystemEnvironment systemEnvironment, Validation<ProductExecutorConfig> validation) {
+        notNull(config, "config may not be null!");
+        notNull(systemEnvironment, "systemEnvironment may not be null!");
+
+        this.config = config;
+        this.systemEnvironment = systemEnvironment;
+
+        if (validation != null) {
+            AssertValidation.assertValid(config, validation);
+        }
+
+        /* create a simple map containing parameters */
+        List<ProductExecutorConfigSetupJobParameter> jobParameters = config.getSetup().getJobParameters();
+        for (ProductExecutorConfigSetupJobParameter jobParameter : jobParameters) {
+            configuredExecutorParameters.put(jobParameter.getKey(), jobParameter.getValue());
+        }
+
+    }
+    
+    public String getPasswordOrAPIToken() {
+        ProductExecutorConfigSetupCredentials credentials = config.getSetup().getCredentials();
+        String pwd = credentials.getPassword();
+        return evaluateEnvironmentEntry(pwd);
+    }
+
+    public String getProductBaseURL() {
+        return config.getSetup().getBaseURL();
+    }
+
+    public String getUser() {
+        ProductExecutorConfigSetupCredentials credentials = config.getSetup().getCredentials();
+        String user = credentials.getUser();
+        return evaluateEnvironmentEntry(user);
+    }
+
+    private String evaluateEnvironmentEntry(String data) {
+        if (data == null) {
+            return null;
+        }
+        if (!data.startsWith(ENV_PREFIX_ID)) {
+            return data;
+        }
+        String key = data.substring(ENV_PREFIX_ID.length());
+        String value = systemEnvironment.getEnv(key);
+        if (value == null) {
+            LOG.warn("No environment entry defined for variable:{}", key);
+        }
+        return value;
+    }
+
+    protected boolean getParameterBooleanValue(String key) {
+        String asText = getParameter(key);
+        return Boolean.parseBoolean(asText);
+    }
+
+    protected String getParameter(String key) {
+        if (key == null) {
+            return null;
+        }
+        return configuredExecutorParameters.get(key);
+    }
+
+    protected int getParameterIntValue(String key) {
+        String asText = getParameter(key);
+        if (asText == null) {
+            return -1;
```

Comment:
Returning `-1` is always strange, is it still is an integer. From my understanding using `Optional` would be better and return `Optional.empty()` instead of `-1`.

Reply:
I want a simple integer and not any Objects here. I am not always a friend of Optional or using always the Integer instead of simple int. 
Reason:  I you are using Objects you always have to check on the other side about null - at least for Optional also in your tests. When using "Integer" as return type you got the situation that when somebody uses this with autoboxing (so as simple int) this can lead to unexpected NPEs inside code.

One thing what whould be better when doing your appraoch : you could use -1 as a "real" value when doing Optional.
But the calling etc. would be ... more pain... if we would keep AdapterConfig and AdapterConfigBuilder parts which are using those.

To mitigate/make usage smooth as currently with int values:
We would need to refactor all config builder patterns and also all config classes for adapters to use Optional<Integer> instead of a simple int...

Reply:
I think a good idea, but we should refactor this in global way - I created a discussion here : https://github.com/Daimler/sechub/discussions/516

Reply:
Here we keep it as is